.goreleaser.yml

```diff
@@ -45,15 +45,16 @@ publishers:
     # Terraform CLI 0.10 - 0.11 perform discovery via HTTP headers on releases.hashicorp.com
     # For providers which have existed since those CLI versions, exclude
     # discovery by setting the protocol version headers to 5.
-    cmd: hc-releases upload-file {{ abs .ArtifactPath }} -header=x-terraform-protocol-version=5 -header=x-terraform-protocol-versions=5.0 -upload-name={{ .ArtifactName }}
+    dir: "{{ dir .ArtifactPath }}"
+    cmd: |
+      hc-releases upload -file={{ .ArtifactName }} -product={{ .ProjectName }} -version={{ .Version }} -header=x-terraform-protocol-version=5 -header=x-terraform-protocol-versions=5.0
```

Comment:
Hi there @bflad -- I made some updates after @jeanneryan + @shore's testing and it should now reflect the use of `-file={{ .ArtifactPath }}`

We had to comment out the `extra_files` bit as it's not currently supported by releases api.

Reply:
Mmm, without uploading `terraform-registry-manifest.json`, how would Terraform CLI know the necessary metadata, especially the protocol?

Reply:
@detro , is it a required file then? I'll bring this up to my team.

Reply:
As I understand it, the "parties" involved are the TF CLI, TF Registry and this file (specifically, the protocol section).

Based on the version of the CLI, a different protocol is necessary, and that information in regards to a specific provider lives in the manifest file.

If that file is not uploaded, the registry doesn't have that information.

The caveat: the content of that manifest seems to be "cached" (for lack of better words) and so, as long as we don't make changes to it (i.e. stick with the same protocol), it will keep working (i.e. we release WITHOUT it and the registry will be able to serve new versions). But when we upgrade the protocol used by the provider, this whole thing breaks because now we are unable to update that protocol information in the registry.

I admit that my understanding is based on observation and conversations with @bflad: so, it's better if he chimes in to correct me if I got something wrong.